packages/apollo-link-http/src/httpLink.ts

```diff
@@ -63,12 +63,19 @@ export const createHttpLink = (linkOptions: HttpLink.Options = {}) => {
     let chosenURI = selectURI(operation, uri);
 
     const context = operation.getContext();
+    const contextHeaders = { ...context.headers };
+
+    if (context.clientAwareness) {
```

Comment:
Are these headers being set at the beginning or the end? I think it's worthwhile documenting override behavior.

Reply:
At the end, so great point @zionts; I'll move them to the beginning and add a small note about overriding them. Thanks!